### ISSUE TITLE
Kontrolisanje simulacije helikoptera pomoću prepoznavanja govora i detekcije šake 

### ISSUE THREAD

**Issue by pavleglusac** [ftn-ai-lab/sc-2022-siit#4]
**Tim**
Pavle Glušac SW 19/2019 

**Asistent**
Milica Škipina

**Definicija problema**
Potrebno je napraviti sistem koji će preko kamere snimati video i na tom videu detektovati poziciju ključnih tačaka na korisnikovoj šaci. Pomoću datih tačaka, potrebno je klasifikovati gesture koji korisnik pravi kao i uglove (roll, pitch i yaw šake).  Takođe je potrebno da sistem može snimati govor preko mikrofona kako bi se omogućilo kontrolisanje pomoću ključnih reči.

**Skup podataka**
Za klasifikovanje gestures šake koristiću skup podataka koji sam sam napravio, dok ću za klasifikaciju govora koristiti [speech commands dataset](https://paperswithcode.com/dataset/speech-commands)

**Metodologija**
Biće napravljena PyQt aplikacija koja će podržavati snimanje govora i videa i imaće konzolu.
U konzoli je moguće izabrati režim (audio/video).

Za video, procedura je sledeća:
1. Uraditi preprocessing slike
2. Sliku proslediti pretreniranom mediapipe modelu koji vraća keypoints šake
3. Uraditi postprocessing keypointsa
4. Proslediti keypoints NN koju sam istrenirao na svom datasetu za gesture
5. Izračunati potrebne uglove
6. Proslediti rezultate podsistemu za upravljanje simulacijom

Za audio, procedura je sledeća
1. Uraditi preprocessing audia
2. Proslediti rezultat CNN modelu koji će biti istreniran nad speech command datasetu
3. Proslediti rezultate podsistemu za upravljanje simulacijom

Svi modeli biće implementirani u PyTorch radnom okviru.

**Evaluacija**
Za svaki od modela koristi se 80-10-10 split podataka i koristi se crossentropy loss. Koristiće se accuracy kao metrika.

**Comment by milica-skipina** [comment]
Tema odobrena.
Ostavite link ka GitHub repozitorijumu projekta.
Srećan rad.

**Comment by pavleglusac** [comment]
https://github.com/pavleglusac/MLAV
